### ISSUE TITLE
Deduplicate cgroup code

### ISSUE THREAD

**Issue by AustinWise** [dotnet/runtime#99171]
It appears this code has been copy-pasted 4 times:

* [src/coreclr/gc/unix/cgroup.cpp](https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/unix/cgroup.cpp)
* [src/coreclr/nativeaot/Runtime/unix/cgroupcpu.cpp](https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/Runtime/unix/cgroupcpu.cpp) 
* [src/coreclr/pal/src/misc/cgroup.cpp](https://github.com/dotnet/runtime/blob/main/src/coreclr/pal/src/misc/cgroup.cpp)
* [src/mono/mono/utils/mono-cgroup.c](https://github.com/dotnet/runtime/blob/main/src/mono/mono/utils/mono-cgroup.c)

I'm guessing a good place to put a consolidated implementation is in [src/native/minipal/](https://github.com/dotnet/runtime/tree/main/src/native/minipal).

**Comment by jkotas** [comment]
cc @janvorli

**Comment by janvorli** [comment]
> I'm guessing a good place to put a consolidated implementation is in [src/native/minipal/](https://github.com/dotnet/runtime/tree/main/src/native/minipal).

Yes, we've already moved the utf8 code there from the coreclr PAL in the past.

I was actually not aware that mono used a clone of that code too. If there are any differences, I believe the files in [src/coreclr/pal/src/misc/cgroup.cpp](https://github.com/dotnet/runtime/blob/main/src/coreclr/pal/src/misc/cgroup.cpp) and [src/coreclr/gc/unix/cgroup.cpp](https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/unix/cgroup.cpp) should be the source of truth.
